src/Models/IoTWorkspaceProject.ts

```diff
@@ -244,18 +241,32 @@ export class IoTWorkspaceProject extends IoTWorkbenchProjectBase {
     await iotHub.load();
     this.componentList.push(iotHub);
 
-    const iothubDevice = new ioTHubDeviceModule.IoTHubDevice(this.channel);
-    this.componentList.push(iothubDevice);
+    const iotHubDevice = new ioTHubDeviceModule.IoTHubDevice(this.projectRootPath, this.channel, [
+      {
+        component: iotHub,

```

Comment:
if we need this kind of structure, we could define it as interface

Reply:
"IoTHubDevice" implements the interfaces "Component" and "Provisionable". Inside "Provisionable" it already has a member "dependencies". The old code didn't pass the dependencies info when creating IoTHubDevice instance. In fact IoTHubDevice should have "IoTHub" as dependency.